### PR TITLE
Grown objects, New food, and old food all grind equally being placed in a grinder from a bag.

### LINKED ISSUE #46356: Novaflowers in a plant bag cannot be placed inside of an all-in-one grinder
Issue reported from  Round ID: 117902 (/tg/Station Sybil [ENGLISH] [US-WEST] [100% LAG FREE])

Reporting client version: 512.1475

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Fixes #46356. What it says on the tin, Grown objects, New food, and old food can all be moved from a tray or plant bag, and mass deposited into a reagent grinder. Anything in those 3 typepaths should all contain at least some kind of reagent content, and if not, is the sort of object to be ground down into nothing if it didn't contain any.
Also, extends newfood and /obj/item/grown parity with other food-adjacent processing tools.

## Why It's Good For The Game

Bugs bad. Improves feature parity between similar reagent-grinding mechanisms and makes processing faster with bags as expected of them.

## Changelog
:cl:
fix: "fancy" new foods and byproducts of grown foods can be mass inserted into reagent grinders as expected.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm [hunk 1]
```diff
@@ -17,6 +17,11 @@
 	var/limit = 10
 	var/speed = 1
 	var/list/holdingitems
+	///What objects will this allow for you to just insert and grind?
+	var/static/list/allowed_grindables = typecacheof(list(
+		/obj/item/food,
+		/obj/item/reagent_containers,
+		/obj/item/grown))
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm:24]: Is this going to be a one time use, or should it be a define like IS_EDIBLE()

Comment (ArcaneMusic) [reply code/modules/reagents/chemistry/machinery/reagentgrinder.dm:24]: This is rather unique in that this is just specifically things that can wind up in a plant bag or a serving tray for fast transfer. The benefit to doing a typecacheof here is that /obj/item/grown is explicitly non-edible, and even before the food refactor that's what caused the linked issue in the first place.

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm [hunk 2]
```diff
@@ -149,15 +154,20 @@
...
+				continue
+			if(holdingitems.len >= limit)
+				to_chat(user, "<span class='warning'>\The [src] is full, you can't put anything in!</span>")
+				return TRUE
+			if(SEND_SIGNAL(T, COMSIG_TRY_STORAGE_TAKE, S, src))
+				loaded++
+				holdingitems += S
+		if(loaded)
+			to_chat(user, "<span class='notice'>You insert [loaded] items into \the [src].</span>")
+		return
```
Comment (LemonInTheDark) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm:170]: A: Why check if the list had items initially? actually why track the list size separately at all? Should it be a counter of how many were added? shouldn't it start at 0 then?
B:does changing the send_signal to the extent that you did effect anything here? tested for feature parity?

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm [hunk 3]
```diff
@@ -17,6 +17,11 @@
 	var/limit = 10
 	var/speed = 1
 	var/list/holdingitems
+	///What objects will this allow for you to just insert and grind?
+	var/static/list/allowed_grindables = typecacheof(list(
+		/obj/item/food,
+		/obj/item/reagent_containers,
```
Comment (Qustinnus) [inline code/modules/reagents/chemistry/machinery/reagentgrinder.dm:23]: why should you be able to grind beakers

Comment (ArcaneMusic) [reply code/modules/reagents/chemistry/machinery/reagentgrinder.dm:23]: I could probably axe reagent containers if #55160 goes through, so I could totally wait for that to go through before this is ready.

File: code/modules/reagents/chemistry/machinery/reagentgrinder.dm (hunk 2 above)
Comment (ArcaneMusic) [reply code/modules/reagents/chemistry/machinery/reagentgrinder.dm:170]: As to B), feature parity is maintained as far as my testing went.
As to A), whoops

**Comment by ArcaneMusic** [comment]
Alright I'm probably just going to need to restart this later, a bit too busy for now to think about this one for the moment.

**Pull request closed without merging**